indev/libinput.c

```diff
@@ -457,50 +554,51 @@ static void read_pointer(libinput_drv_state_t *state, struct libinput_event *eve
 static void read_keypad(libinput_drv_state_t *state, struct libinput_event *event) {
   struct libinput_event_keyboard *keyboard_event = NULL;
   enum libinput_event_type type = libinput_event_get_type(event);
+  libinput_lv_event_t *evt = NULL;
```

Comment:
This doesn't appear to ever be set to something else than `NULL`. I think you're missing an `evt = new_event(state);`  inside the `case LIBINPUT_EVENT_KEYBOARD_KEY` below?